5.3.Expense Tracking Frontend Implementation (View Budgets & View Budget Details)

RynniaRyan opened the pull request:
## **Budget Screen Preview** 
I have added new components and re-arranged the layout of the Budget screen see pic as preview

<img width="170" alt="image" src="https://github.com/user-attachments/assets/c70127c9-d6a8-4fe4-9cd8-d23db2b20914" /> 
<img width="169" alt="image" src="https://github.com/user-attachments/assets/7a989241-f703-4e0c-bc2b-ad7eab8f81dd" />


### **However!** 
I did this using dummy data placeholder as all the user data was not being fetch (same with Transactions). I believe that this error is being displayed within BudgetScreen.js and not budget.py in views route
  
 I've tried fixing but in the lack of time I decided to use placeholder to work with in the meanwhile. Hence the Draft Pull Request
 
### Edit!
 Ok so I just found out that the data is not being fetched on the mobile only on the web version of the application..hmm

RynniaRyan committed:
Added NotificationBell, FilterTags & Arranged Layout

Added a notification bell with pop up model and filter tag component.

I could not work with the rendered user data so I had to work with 'dummy data' for the time being.

The filter tags are currently configured to filter by budget title for now as placeholder for 'individual' & 'family' type budgets

JaleneArmstrong reviewed (changes requested):
![Screenshot 2025-02-20 211233](https://github.com/user-attachments/assets/0902f078-e0b1-4400-96c7-7d4e3121e7a5)

If you look at the fetchData function, after it collects all the budgets, it uses  setData(budgets);, this populates the data const here const [data, setData] = useState([]); instead of using your dummyBudgets const, all you had to do was replace that with the data const.

const filteredBudgets = selectedFilter === 'All'
    ? data
    : data.filter((budget) => budget.budgetTitle === selectedFilter);

Since you restrained the budgets showing up to having to first obey the filter, you gotta use the const that actually holds the budgets, so it can show the budgets.

There’s also a design error in your code, the filter is based on the budgetTitle, but not everyone is gonna name their budget “Entertainment” or “Rent”.

RynniaRyan committed:
Attempt To Fix Budget Fetch Failure

I took the advice given to me by my peer, however there was still failure in retrieving user budgets on my end. :(

Ref: https://github.com/Chroma-Corps/FFM-Application/pull/18#pullrequestreview-2631602824

RynniaRyan commented:
> If you look at the fetchData function, after it collects all the budgets, it uses  setData(budgets);, this populates the data const here const [data, setData] = useState([]); instead of using your dummyBudgets const, all you had to do was replace that with the data const.

### Edit (2):
So i did change the values but as expected it still has trouble fetching the data, but at least it works on your end. Once again its always some failure to connect to backend stuff on my end 😞 .

I Committed my attempt to fix like you suggested. I also made the Filtertags more logical by changing including 'budgetType' within Budget related code in  model, views and controller

RynniaRyan committed:
Fixed Functionality Of FilterTags With DummyData

As suggested my team member, I accommodated a way for the filter tags to filter out the diff between individual and family budgets.

I had to make adjustments to the Backend to include 'budgetType' attribute as well as all the budget creation related functions and routes.

However I still had to work with the dummy data as my user data is not being fetched :(

JaleneA commented:
![](https://github.trello.services/images/mini-trello-icon.png) [5.3. Expense Tracking Frontend Implementation (View Budgets)](https://trello.com/c/A57L9CgB/54-53-expense-tracking-frontend-implementation-view-budgets)

RynniaRyan committed:
Fixed Data Fetching (Web)

I fixed the issue relating to failed data fetching as well as edited the initialize.py to have a longer list of budgets for user 'Bob'

RynniaRyan committed:
Base For Editing Without Expo Go :(

Since my Expo Go app is giving issues, I'll be using the direct route for fetching user data so I won't have to keep logging in over and over.

I ill revert it back to ho it as originally after I'm finished

RynniaRyan committed:
Implemented Progress Bar & Edited Budget Card Layout

Note : i disabled all Authorization/Token related requirements for fetching budgets since on mu end fetching them is an issue.

If your Expo Go works simply just uncomment
lines 42-46 within BudgetScreen.js and  line 52 in budget(views)

RynniaRyan committed:
Worked on BudgetDetailsScreen

List of main tasks done on this screen:

- Adjusted layout
- Allowed Scrollable Transaction View
- Created/Added Circular Graph Component (Not functioning Properly)

Ref Detailed List of Tasks Here: https://trello.com/c/A57L9CgB/54-53-expense-tracking-frontend-implementation-view-budgets

RynniaRyan committed:
Reverted App Screen Stack & Fixed Backend RemainingAmmount Attribute

I Forgot to revert changes made on App.js and adjustments made to fetch related code.

I also edited the backend specifically add_transaction controller to handle updating budget's remaining amount if tied to a budget

RynniaRyan commented:
### **Pull Request Update!** 
 I edited both the Budgets View & Budget Details View hence the renaming of the branch

### Pending Fixes:
- Proper functionality of Pie chart
- Addition of Graph key Index

## Screen Previews:
<img width="167" alt="image" src="https://github.com/user-attachments/assets/edb00c96-d3f5-4299-9421-fc33c3940c55" />
<img width="168" alt="Screenshot 2025-02-28 122631" src="https://github.com/user-attachments/assets/fb5121fe-a5b7-4971-834c-f48453448781" />
